pkg/controller/volume/persistentvolume/pv_controller.go

```diff
@@ -583,12 +588,17 @@ func (ctrl *PersistentVolumeController) syncVolume(volume *v1.PersistentVolume)
 					return err
 				}
 			}
-
 			if err = ctrl.reclaimVolume(volume); err != nil {
 				// Release failed, we will fall back into the same condition
 				// in the next call to this method
 				return err
 			}
+			if volume.Spec.PersistentVolumeReclaimPolicy == v1.PersistentVolumeReclaimRetain {
```

Comment:
This is currently normal behavior a volume in "Released" phase. I don't think we need to post an event for this.

Regarding your question, we intentionally do not unbind the volume here. The idea behind Retain is that an administrator may want to clean up the disk before making it available to users again. So once an administrator has confirmed that the disk is ready to be reused, then they can explicitly clear the ClaimRef, which will make it "Available" again.